src/scripts/browserDeviceProfile.js

```diff
@@ -318,8 +318,14 @@ import browser from './browser';
         }
 
         const isSurroundSoundSupportedBrowser = browser.safari || browser.chrome || browser.edgeChromium || browser.firefox || browser.tv || browser.ps4 || browser.xboxOne;
+        const isAc3Eac3Supported = supportsAc3(videoTestElement) || supportsEac3(videoTestElement);
         const speakerCount = getSpeakerCount();
 
+        // AC3/EAC3 hinted that device is able to play dolby surround sound.
+        if (isAc3Eac3Supported && isSurroundSoundSupportedBrowser) {
+            return 6;
+        }
+
```

Comment:
Rearrange to remove unnecessary call of `getSpeakerCount`.
```suggestion
        const isAc3Eac3Supported = supportsAc3(videoTestElement) || supportsEac3(videoTestElement);

        // AC3/EAC3 hinted that device is able to play dolby surround sound.
        if (isAc3Eac3Supported && isSurroundSoundSupportedBrowser) {
            return 6;
        }

        const speakerCount = getSpeakerCount();

```